### ISSUE TITLE
Split spec helpers into logical files

### ISSUE THREAD

**Issue by voodoorai2000** [consul/consul#2644]
Context
===
In order to make specs more legible [helpers](https://github.com/consul/consul/blob/master/spec/support/common_actions.rb) are used in the codebase
These helpers have grown large and are a little hard to follow

What
===
Split the spec helper file `common_actions.rb` into logical files:
- spec/support/budgets.rb
- spec/support/proposals.rb
- spec/support/comments.rb
- spec/support/tags.rb
- spec/support/votes.rb
- spec/support/verifications.rb
- ...

Notes
===
A good rule of thumb is to reproduce the file structure in app/models 👌

**Comment by aamarill** [comment]
@voodoorai2000 Can I take this one? 💪 😎

**Comment by voodoorai2000** [comment]
Yes @aamarill 😌👏🏻🎉💪🏻💫

**Comment by voodoorai2000** [comment]
Closed via https://github.com/consul/consul/pull/2653
